tools/noopfactory/src/main/kotlin/com/datadog/tools/noopfactory/NoOpFactoryAnnotationProcessor.kt

```diff
@@ -87,13 +88,24 @@ class NoOpFactoryAnnotationProcessor : AbstractProcessor() {
         val packageName = typeElement.qualifiedName.toString().substringBeforeLast('.')
         val typeSpec = generateTypeSpec(typeElement, packageName)
 
-        val file = FileSpec.builder(packageName, typeSpec.name.orEmpty())
+        val className = typeSpec.name.orEmpty()
+        val file = FileSpec.builder(packageName, className)
             .addType(typeSpec)
+            .indent("    ")
             .build()
 
-        file.writeTo(File(kaptKotlinGeneratedDir))
+        val kotlinFileObject = processingEnv.filer
```

Comment:
nice....where did you get that ?

Reply:
in the annotation processing documentation